### PR TITLE
Add category to communities 

### LINKED ISSUE #12066: Add category to communities
As an admin
I want to be able to add categories to my community
So that I can group channels that are somehow related

### Description

The admin should be able to create "categories" and include channels in them.
Categories are essentially folders where an admin can group channels.

This task includes all the management of categories and channels, it's fairly chunky, and it can be broken up if necessary.

### Designs

https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=16052%3A2354

### PR DESCRIPTION
**Pull request by flexsurfer**
fixes: #12066

items reordering is not implemented in this PR https://github.com/status-im/status-react/issues/12292
items persistence is not implemented https://github.com/status-im/status-react/issues/12293

### CODE REVIEW

**Comment by churik** [comment]
@flexsurfer 
### ISSUE 1: can't see categories until adding 1 channel

**Steps:**
1) restore multiaccount from seed phrase
2) create a community
3) add category

**Expected result:**
can see community

**Actual result:**

https://user-images.githubusercontent.com/4557972/125422555-d11d43ad-983b-4e98-a76a-3eb0b45197c2.mp4


**OS:** IOS, Android

### ISSUE 2: no option to delete / edit community
From design I can see only delete: https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=16052%3A3955
@errorists maybe I missed designs for editing community category?

**Comment by errorists** [comment]
Apologies, editing is found in the newest part of the Figma where I revised all screens to meet Phase 3 requirements. [🔗 Here](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=21267%3A98474)

**Comment by flexsurfer** [comment]
ISSUE 2: no option to delete / edit community
there is edit chats option under "..." in community

**Comment by flexsurfer** [comment]
thanks @churik  fixed

**Comment by churik** [comment]
@flexsurfer thank you!

### ISSUE 3: last channel name is shown and group chat icons instead of public when adding new category

**Steps:**
1) restore multiaccount from seed phrase
2) add community, add channel
3) tap on "Create category"

Monosnap 2021-07-19 12-44-00

**OS:** IOS, Android

### ISSUE 4: no option to add chat into category when creating a channel
Design in [here](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=20765%3A2732)

### ISSUE 5: No options when long-press category
On "Edit chats" interface there is option to delete categories or to delete chats from categories.
But there is no option to rename category, as it should be according to the [design](https://www.figma.com/file/aS1ct66VQ6V0cio7vSqS8UoG/Chat?node-id=21267%3A98474) 

### ISSUE 6: (not related to PR, but to navigation, maybe you can fix it in scope) "Back" from community channel redirects to home instead of community page

File: src/status_im/ui/screens/communities/community.cljs [hunk 1]
```diff
@@ -137,23 +154,49 @@
...
+     [:<>
+      [accordion/section
+       {:opened   edit
+        :disabled edit
+        :title    [rn/view styles/category-item
+                   (if edit
+                     [rn/touchable-opacity {:on-press #(>evt [:delete-community-category community-id id])}
+                      [icons/icon :main-icons/delete-circle {:no-color true}]]
+                     [icons/icon :main-icons/channel-category {:color colors/gray}])
+                   [rn/text {:style {:font-size 17 :margin-left 10}} name]]
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/communities/community.cljs:170]: # 🧚🏼‍♀️
Magic number 17 `:font-size 17`,  maybe use `quo/text`?

File: src/status_im/ui/screens/communities/select_category.cljs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+            [status-im.utils.handlers :refer [>evt <sub]]
+            [status-im.ui.components.topbar :as topbar]
+            [status-im.ui.components.icons.icons :as icons]
+            [status-im.ui.components.colors :as colors]
+            [status-im.communities.core :as communities]))
+
+(def selected-item (reagent/atom ""))
+
+(defn render-fn [{:keys [name id]}]
+  (println id @selected-item)
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/communities/select_category.cljs]: Rogue `println`

**Review by shivekkhurana — APPROVED** [review]

**Comment by churik** [comment]
@flexsurfer not sure, maybe ISSUE 4 and ISSUE 5 are part of `phase 2` or `phase 3`

**Comment by flexsurfer** [comment]
thanks @churik ISSUE 3 fixed

**Comment by churik** [comment]
Thank you! Issue 3 is fixed and verified, leftovers will be reported separately.
One more small issue: category is available for changing to non-admin user, but in fact he can't finish process (nothing happens on tapping "Done" button)
photo_2021-07-21 13 10 51 jpeg 2021-07-21 13-11-24

**Comment by flexsurfer** [comment]
thanks @churik fixed

**Pull request merged**